### DISCUSSION TITLE
Zigbee2MQTT Post Install | USB-Adapter

### DISCUSSION THREAD

**Discussion by MickLesk** [community-scripts/ProxmoxVE#410] [Guides]
Copied from original post: https://github.com/tteck/Proxmox/discussions/2917

# Zigbee2MQTT Setup Guide

This guide will help you set up Zigbee2MQTT on your system. Follow the steps carefully to ensure a smooth installation and configuration process.

## ⚙️ Determine the Location of Your Adapter

Run the following command to find the location of your Zigbee adapter:

```bash
ls -l /dev/serial/by-id
```

**Example Output:**
```bash
lrwxrwxrwx 1 root root 13 Jun 19 17:30 usb-1a86_USB_Serial-if00-port0 -> ../../ttyUSB0
```
⚠️Make a note of the value after -> (e.g., /dev/ttyUSB0 or /dev/serial/by-id/usb-...), as you will need it in the configuration file.

## ⚙️ Edit the Configuration File
Before starting Zigbee2MQTT, edit the configuration.yaml file.

📃**For Debian:**
```bash
nano /opt/zigbee2mqtt/data/configuration.yaml
```

📃**For Alpine:** 
```bash
nano /etc/zigbee2mqtt/configuration.yaml
```

Save and exit the editor using: Ctrl+O, then Enter, and Ctrl+X.

## 🔏 Example Configuration:

```yaml
frontend:
  port: 9442
homeassistant: true
permit_join: false
mqtt:
  base_topic: zigbee2mqtt
  server: 'mqtt://192.168.86.224:1883'
  user: usr
  password: pwd
  keepalive: 60
  reject_unauthorized: true
  version: 4
serial:
  # Location of the adapter
  # USB adapters - use format "port: /dev/serial/by-id/XXX"
  # Ethernet adapters - use format "port: tcp://192.168.1.12:6638"
  port: /dev/serial/by-id/usb-Texas_Instruments_TI_CC2531_USB_CDC___0X00124B0018ED3DDF-if00
  # Adapter type, allowed values: `zstack`, `ember`, `deconz`, `zigate` or `zboss`
  adapter: zstack
advanced:
  pan_id: GENERATE
  network_key: GENERATE
  channel: 20
```

Make sure to update the serial section with your adapter's location and uncomment the appropriate adapter type if necessary.

## ⚙️ Start Zigbee2MQTT
After completing the configuration, you can start Zigbee2MQTT.

📃**For Debian:**
```bash
systemctl restart zigbee2mqtt
```

📃**For Alpine:** 
```bash
rc-update add zigbee2mqtt default
```

**Comment by dsiebel** [comment]
I'd like to point out two things here:
1. The container comes with an annotated example config at `opt/zigbee2mqtt/data/configuration.example.yaml` which contains all the explanation needed to get started (imo).
2. The Debian container uses a systemd service, so it should be `systemctl start zigbee2mqtt.service`.

**Comment by JesperKuipers** [comment]


**Reply by JesperKuipers** [reply]
dev branch seems to work tho

**Reply by D3rPaPaH0d3n** [reply]
How did you change to dev branch? I tried already so many things, but i don´t get it!

**Comment by LeonGrim2024** [comment]
I have not any errors, but can't connect
![post install z2m in lxc](https://github.com/user-attachments/assets/1e875bec-5c22-4ae6-b835-3fcede742f92)
 to web-interface after install. and console look odd. reboot not fix. HELP...

**Reply by dsiebel** [reply]
Did you enable the frontend in your config?
https://www.zigbee2mqtt.io/guide/configuration/frontend.html

If the frontend was enabled, the log should show a line like:
```
info: z2m: Started frontend on port 0.0.0.0:8080
```

**Comment by hadros01** [comment]
Hello,
I'm a little bit lost. I followed (hopefully complete) the documentation to install zigbee2mqtt on a raspberry 3b. I use the sonoff zigbee 3.0 USB dongle plus.While the installation I didn't saw any error. Now, if I start  it, I get the following log and after that the raspi stuck and is dead. Not reachable anymore. Following the log infos.

> zigbee2mqtt@1.42.0 start
> node index.js

Starting Zigbee2MQTT without watchdog.
[2024-12-02 22:31:32] info: 	z2m: Logging to console, file (filename: log.log)
[2024-12-02 22:31:32] debug: 	z2m: Can't load state from file /opt/zigbee2mqtt/data/state.json (doesn't exist)
[2024-12-02 22:31:32] info: 	z2m: Starting Zigbee2MQTT version 1.42.0 (commit #861cba6)
[2024-12-02 22:31:32] debug: 	z2m: sd-notify loaded
[2024-12-02 22:31:32] info: 	z2m: Starting zigbee-herdsman (2.1.9)
[2024-12-02 22:31:32] debug: 	z2m: Using zigbee-herdsman with settings: '"{\"network\":{\"panID\":6754,\"extendedPanID\":[221,221,221,221,221,221,221,221],\"channelList\":[11],\"networkKey\":\"HIDDEN\"},\"databasePath\":\"/opt/zigbee2mqtt/data/database.db\",\"databaseBackupPath\":\"/opt/zigbee2mqtt/data/database.db.backup\",\"backupPath\":\"/opt/zigbee2mqtt/data/coordinator_backup.json\",\"serialPort\":{\"path\":\"/dev/serial/by-id/usb-ITEAD_SONOFF_Zigbee_3.0_USB_Dongle_Plus_V2_20230329145849-if00\",\"adapter\":\"ember\"},\"adapter\":{\"disableLED\":false}}"'
[2024-12-02 22:31:33] debug: 	zh:adapter: Path '/dev/serial/by-id/usb-ITEAD_SONOFF_Zigbee_3.0_USB_Dongle_Plus_V2_20230329145849-if00' is valid for 'EmberAdapter'
[2024-12-02 22:31:33] info: 	zh:ember: Using default stack config.
[2024-12-02 22:31:33] debug: 	zh:controller: Starting with options '{"network":{"networkKeyDistribute":false,"networkKey":"HIDDEN","panID":6754,"extendedPanID":[221,221,221,221,221,221,221,221],"channelList":[11]},"serialPort":{"path":"/dev/serial/by-id/usb-ITEAD_SONOFF_Zigbee_3.0_USB_Dongle_Plus_V2_20230329145849-if00","adapter":"ember"},"adapter":{"disableLED":false},"databasePath":"/opt/zigbee2mqtt/data/database.db","databaseBackupPath":"/opt/zigbee2mqtt/data/database.db.backup","backupPath":"/opt/zigbee2mqtt/data/coordinator_backup.json"}'
[2024-12-02 22:31:33] info: 	zh:ember: ======== Ember Adapter Starting ========
[2024-12-02 22:31:33] info: 	zh:ember:ezsp: ======== EZSP starting ========
[2024-12-02 22:31:33] info: 	zh:ember:uart:ash: ======== ASH Adapter reset ========
[2024-12-02 22:31:33] info: 	zh:ember:uart:ash: RTS/CTS config is off, enabling software flow control.
[2024-12-02 22:31:33] debug: 	zh:ember:uart:ash: Opening serial port with {"path":"/dev/serial/by-id/usb-ITEAD_SONOFF_Zigbee_3.0_USB_Dongle_Plus_V2_20230329145849-if00","baudRate":115200,"rtscts":false,"autoOpen":false,"parity":"none","stopBits":1,"xon":true,"xoff":true}

**Reply by dsiebel** [reply]
You mention installing on a Raspberry Pi, how is this related to Proxmox LXC and this script?

**Comment by hadros01** [comment]
Yes, I made the installation related to the description for Linux Raspian. From Proxmox LXC I know nothing. Why do you ask about Proxmox?

**Reply by dsiebel** [reply]
Because you're posting in a repository that provides script to provision Proxmox LXCs and VMs.
This repository has nothing to do with Raspbian OR Raspberry Pi.

**Comment by hadros01** [comment]
Oh sorry. I'm new here in Git and it is a little bit unclear to me. 
But nevertheless I have solved my problem.

**Comment by Kuchiru** [comment]
Hello there,

How do i update/upgrade Z2M on the Alpine version of this LXC?

**Reply by Kuchiru** [reply]
Appears it is installed as a package `apk update && apk upgrade` will do it.

**Reply by JensjJ8200** [reply]
how do you update the non Alpine version?

**Reply by Kuchiru** [reply]
> how do you update the non Alpine version?

rerun the install script inside the vm/lxc, it'll give you an upgrade option.

**Comment by alvariyo** [comment]
hello,
I am facing a blocking issue that I believe that is on the mosquitto part, but not able to realize what it is.
I have test everything diferent config version, single quotes... can anyone help pls?

My configuration:
frontend:
  port: 9442
  homeassistant: true
  permit_join: false
mqtt:
  base_topic: zigbee2mqtt
  server: 'mqtt://192.168.1.250:1883'
  user: 'ust'
  password: 'pass'
  keepalive: 60
  reject_unauthorized: true
  version: 3
serial:
  port: /dev/serial/by-id/usb-ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_d27dd0fce91bef119293a3d94909ffd0-if00-port0
  adapter: zstack
advanced:
  pan_id: 13187
  network_key:
    - 209
    - 91
    - 170
    - 133
    - 82
    - 203
    - 65
    - 98
    - 105
    - 176
    - 171
    - 146
    - 250
    - 251
    - 152
    - 209
  channel: 14


Logs:
Dec 16 02:26:04 zigbee2mqtt npm[6890]: > node index.js
Dec 16 02:26:04 zigbee2mqtt npm[6903]: Starting Zigbee2MQTT without watchdog.
Dec 16 02:26:05 zigbee2mqtt npm[6903]: [2024-12-16 02:26:05] info:         z2m: Logging to console, file (filename: log.log)
Dec 16 02:26:05 zigbee2mqtt npm[6903]: [2024-12-16 02:26:05] info:         z2m: Starting Zigbee2MQTT version 1.42.0 (commit #bc6eccf)
Dec 16 02:26:05 zigbee2mqtt npm[6903]: [2024-12-16 02:26:05] info:         z2m: Starting zigbee-herdsman (2.1.9)
Dec 16 02:26:05 zigbee2mqtt npm[6903]: [2024-12-16 02:26:05] info:         zh:zstack:znp: Opening SerialPort with {"path":"/dev/serial/by-id/usb-ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_d27dd0fce91bef119293a3d94909ffd0-if00-port0","baudRate":115200,"rtsctsUSB_Dongle_Plus_d27dd0fce91bef119293a3d94909ffd0-if00-port0","baudRate":115200,"rtscts":false,"autoOpen":false}
Dec 16 02:26:05 zigbee2mqtt npm[6903]: [2024-12-16 02:26:05] info:         zh:zstack:znp: Serialport opened
Dec 16 02:26:15 zigbee2mqtt systemd[1]: Stopping zigbee2mqtt.service - zigbee2mqtt...
Dec 16 02:26:15 zigbee2mqtt npm[6903]: [2024-12-16 02:26:15] error:         z2m: Failed to call 'Frontend' 'stop' (TypeError: Cannot read properties of undefined (Reading 'close')
Dec 16 02:26:15 zigbee2mqtt npm[6903]:     at /opt/zigbee2mqtt/lib/extension/frontend.ts:126:52
Dec 16 02:26:15 zigbee2mqtt npm[6903]:     at new Promise (<anonymous>)
Dec 16 02:26:15 zigbee2mqtt npm[6903]:     at Frontend.stop (/opt/zigbee2mqtt/lib/extension/frontend.ts:126:15)
Dec 16 02:26:15 zigbee2mqtt npm[6903]:     at Controller.callExtensions (/opt/zigbee2mqtt/lib/controller.ts:399:17)
Dec 16 02:26:15 zigbee2mqtt npm[6903]:     at Controller.stop (/opt/zigbee2mqtt/lib/controller.ts:265:9)
Dec 16 02:26:15 zigbee2mqtt npm[6903]:     at stop (/opt/zigbee2mqtt/index.js:165:5)
Dec 16 02:26:15 zigbee2mqtt npm[6903]:     at process.handleQuit (/opt/zigbee2mqtt/index.js:173:13))
Dec 16 02:26:15 zigbee2mqtt npm[6903]: [2024-12-16 02:26:15] error:         z2m: Not connected to MQTT server!
Dec 16 02:26:15 zigbee2mqtt npm[6903]: [2024-12-16 02:26:15] error:         z2m: Cannot send message: topic: 'zigbee2mqtt/bridge/state', payload: 'offline
Dec 16 02:26:15 zigbee2mqtt npm[6903]: [2024-12-16 02:26:15] info:         z2m: Disconnecting from MQTT server
Dec 16 02:26:15 zigbee2mqtt npm[6903]: [2024-12-16 02:26:15] info:         z2m: Stopping zigbee-herdsman...
Dec 16 02:26:15 zigbee2mqtt npm[6903]: [2024-12-16 02:26:15] info:         z2m: Stopped zigbee-herdsman
Dec 16 02:26:15 zigbee2mqtt npm[6903]: [2024-12-16 02:26:15] info:         z2m: Stopped Zigbee2MQTT
Dec 16 02:26:15 zigbee2mqtt systemd[1]: zigbee2mqtt.service: Deactivated successfully.

Any idea?
Many thanks in advance

**Reply by dsiebel** [reply]
Looks like the MQTT server can not be connected.
Have you tried connecting to `mqtt://192.168.1.250:1883` with e.g. MQTT Explorer? Did it work?
Have you tried removing the `mqtt://`? Shouldn't be needed since it's implied.

**Comment by alvariyo** [comment]
hello,

thanks for ur help I tried also but unfortunately still not working.

Any idea?

**Reply by MickLesk** [reply]
https://github.com/Koenkk/zigbee2mqtt/issues/16207

thats not an support discussion!

**Reply by dsiebel** [reply]
yeah, so the config is fine. Looks like you need to update your version of zigbee2mqtt.

**Comment by JGSBP2S** [comment]


**Reply by MickLesk** [reply]
this is not a discussion!

**Comment by nasm27** [comment]
Hello,

I'm trying to update zigbee2mqtt debian to version 2.0.0 but I got error, and also if I try to build new container I got error:
![image](https://github.com/user-attachments/assets/d66affcc-e867-498a-be5b-25363d9524f4)

Please, What is you procedure for update?

Regards

**Reply by MickLesk** [reply]
Definitly wrong script and read changelog With breaking Change & this is not n Issue topic

**Reply by nasm27** [reply]
Thank you.
What changlog you refer? I will read it for sure. Because this is stuck, and the only way I see is moving to ZHA. But I prefer this current method.

**Reply by MickLesk** [reply]
Our Changelog. & U use the tteck repo

**Comment by Beel12213** [comment]
Hey, I'm having real trouble getting this to work for some reason and I'm not sure where I'm going wrong. I've used the Proxmox Helper Script to setup and install the LXC. Noticed the USB device is automatically setup/passed through, so I've then gone ahead and edited the config as suggested in your guide here.

However, for the life of me, I can't figure out why the frontend is not booting up.

When I looked at the logs I got the below error, is anyone able to help me figure out where I'm going wrong please?
```shell
Jan 16 10:58:24 zigbee2mqtt systemd[1]: Started zigbee2mqtt.service - zigbee2mqtt.
Jan 16 10:58:24 zigbee2mqtt pnpm[2361]: > zigbee2mqtt@2.0.0 start /opt/zigbee2mqtt
Jan 16 10:58:24 zigbee2mqtt pnpm[2361]: > node index.js
Jan 16 10:58:25 zigbee2mqtt pnpm[2373]: Starting Zigbee2MQTT without watchdog.
Jan 16 10:58:25 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:25] info:         z2m: Logging to console, file (filename: log.log)
Jan 16 10:58:25 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:25] info:         z2m: Starting Zigbee2MQTT version 2.0.0 (commit #unknown)
Jan 16 10:58:25 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:25] info:         z2m: Starting zigbee-herdsman (3.2.1)
Jan 16 10:58:25 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:25] info:         zh:ember: Using default stack config.
Jan 16 10:58:25 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:25] info:         zh:ember: ======== Ember Adapter Starting ========
Jan 16 10:58:25 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:25] info:         zh:ember:ezsp: ======== EZSP starting ========
Jan 16 10:58:25 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:25] info:         zh:ember:uart:ash: ======== ASH Adapter reset ========
Jan 16 10:58:25 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:25] info:         zh:ember:uart:ash: RTS/CTS config is off, enabling software flow control.
Jan 16 10:58:25 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:25] info:         zh:ember:uart:ash: Serial port opened
Jan 16 10:58:25 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:25] info:         zh:ember:uart:ash: ======== ASH starting ========
Jan 16 10:58:28 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:28] info:         zh:ember:uart:ash: ======== ASH Adapter reset ========
Jan 16 10:58:28 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:28] info:         zh:ember:uart:ash: ======== ASH starting ========
Jan 16 10:58:30 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:30] info:         zh:ember:uart:ash: ======== ASH Adapter reset ========
Jan 16 10:58:30 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:30] info:         zh:ember:uart:ash: ======== ASH starting ========
Jan 16 10:58:33 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:33] info:         zh:ember:uart:ash: ======== ASH Adapter reset ========
Jan 16 10:58:33 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:33] info:         zh:ember:uart:ash: ======== ASH starting ========
Jan 16 10:58:35 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:35] info:         zh:ember:uart:ash: ======== ASH Adapter reset ========
Jan 16 10:58:35 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:35] info:         zh:ember:uart:ash: ======== ASH starting ========
Jan 16 10:58:38 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:38] error:         z2m: Error while starting zigbee-herdsman
Jan 16 10:58:38 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:38] error:         z2m: Failed to start zigbee-herdsman
Jan 16 10:58:38 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:38] error:         z2m: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start_crashes-runtime.html for possible solutions
Jan 16 10:58:38 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:38] error:         z2m: Exiting...
Jan 16 10:58:38 zigbee2mqtt pnpm[2373]: [2025-01-16 10:58:38] error:         z2m: Error: Failed to start EZSP layer with status=HOST_FATAL_ERROR.
Jan 16 10:58:38 zigbee2mqtt pnpm[2373]:     at EmberAdapter.initEzsp (/opt/zigbee2mqtt/node_modules/.pnpm/zigbee-herdsman@3.2.1/node_modules/zigbee-herdsman/src/adapter/ember/adapter/emberAdapter.ts:670:19)
Jan 16 10:58:38 zigbee2mqtt pnpm[2373]:     at EmberAdapter.start (/opt/zigbee2mqtt/node_modules/.pnpm/zigbee-herdsman@3.2.1/node_modules/zigbee-herdsman/src/adapter/ember/adapter/emberAdapter.ts:1533:24)
Jan 16 10:58:38 zigbee2mqtt pnpm[2373]:     at Controller.start (/opt/zigbee2mqtt/node_modules/.pnpm/zigbee-herdsman@3.2.1/node_modules/zigbee-herdsman/src/controller/controller.ts:136:29)
Jan 16 10:58:38 zigbee2mqtt pnpm[2373]:     at Zigbee.start (/opt/zigbee2mqtt/lib/zigbee.ts:69:27)
Jan 16 10:58:38 zigbee2mqtt pnpm[2373]:     at Controller.start (/opt/zigbee2mqtt/lib/controller.ts:142:13)
Jan 16 10:58:38 zigbee2mqtt pnpm[2373]:     at start (/opt/zigbee2mqtt/index.js:161:5)
Jan 16 10:58:38 zigbee2mqtt pnpm[2361]:  ELIFECYCLE  Command failed with exit code 1.
Jan 16 10:58:38 zigbee2mqtt systemd[1]: zigbee2mqtt.service: Main process exited, code=exited, status=1/FAILURE
Jan 16 10:58:38 zigbee2mqtt systemd[1]: zigbee2mqtt.service: Failed with result 'exit-code'.
Jan 16 10:58:38 zigbee2mqtt systemd[1]: zigbee2mqtt.service: Consumed 1.753s CPU time.
Jan 16 10:58:38 zigbee2mqtt systemd[1]: zigbee2mqtt.service: Scheduled restart job, restart counter is at 37.
Jan 16 10:58:38 zigbee2mqtt systemd[1]: Stopped zigbee2mqtt.service - zigbee2mqtt.
Jan 16 10:58:38 zigbee2mqtt systemd[1]: zigbee2mqtt.service: Consumed 1.753s CPU time.
``` 

My config file is below - I'm using the device "SONOFF Universal Zigbee 3.0 USB Dongle Plus Gateway with Antenna for Home Assistant, IoBroker, Zigbee2MQTT":
```
frontend:
  enabled: true
  port: 9442
homeassistant:
  enabled: true
mqtt:
  base_topic: zigbee2mqtt
  server: mqtt://192.168.1.36:1883
  user: usr
  password: pwd
  keepalive: 60
  reject_unauthorized: true
  version: 4
serial:
  port: >-
    /dev/serial/by-id/usb-ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_b071257761beed11b7dc9b4838a92db5-if00-port0
  adapter: ember
advanced:
  pan_id: 39150
  network_key:
    - 174
    - 44
    - 17
    - 18
    - 62
    - 66
    - 151
    - 66
    - 34
    - 219
    - 217
    - 171
    - 189
    - 15
    - 138
    - 217
  channel: 20
version: 4
```

**Reply by MickLesk** [reply]
Error: Failed to start EZSP layer with status=HOST_FATAL_ERROR.

check if your stick reachable or connected

by the way, you have an Sonoff, then its zstack?

**Reply by nasm27** [reply]
The USB device is not in use by other service? or container? Reboot the proxmox it self.
I know if I stop one zigbee2mqtt LXC and try to create a new one I got that error. I need to stop the old LXC present it to start on boot and then reboot. Like that I can create a new one. Was my story....

**Reply by Beel12213** [reply]
Used both fixes suggested above and it seems to be working/booting up now. Really apprecaite both your help and QUICK feedback.

**Steps to Fix (incase anyone else has this issue):**
1. Delete the LXC
2. Reboot Proxmox
3. Create the LXC (Don't manually passthrough the USB, it's done automatically with the script!)
4. Make sure to select the right adapter type (ember/zstack/deconz/zigate/zboss)
5. Follow the instructions from the main post
6. (The MQTT section wants your broker's details not the zigbee2mqtt's IP. Don't be dumb x 2 like me)

**Comment by JensjJ8200** [comment]


**Reply by MickLesk** [reply]
Ember is wrong for sonoff? Zstack should be right. My 2 old sonoffs where anytime zstack? Or is this a special flashed ?

I think the Z2M Repo where better to ask

**Reply by MickLesk** [reply]
I only see, that the adapter crashed at start, but idk, ive now an Network device (POE) for this

**Comment by tugsi** [comment]
@MickLesk Please edit the Configuration Example under the Point Serial is the type of Adapter to be declared
```C
serial:
    # Location of the adapter
    # USB adapters - use format "port: /dev/serial/by-id/XXX"
    # Ethernet adapters - use format "port: tcp://192.168.1.12:6638"
    port: /dev/serial/by-id/usb-Texas_Instruments_TI_CC2531_USB_CDC___0X00124B0018ED3DDF-if00
    # Adapter type, allowed values: `zstack`, `ember`, `deconz`, `zigate` or `zboss`
    adapter: zstack
``` 

Found on [https://github.com/Koenkk/zigbee2mqtt.io/blob/dev/docs/guide/configuration/adapter-settings.md#basic-configuration](https://github.com/Koenkk/zigbee2mqtt.io/blob/dev/docs/guide/configuration/adapter-settings.md#basic-configuration)

**Reply by MickLesk** [reply]
done

**Comment by LeonGrim2024** [comment]
how to update to Zigbee2MQTT V2 now ?

**Reply by MickLesk** [reply]
read changelog and guides not here!

**Reply by tugsi** [reply]
> how to update to Zigbee2MQTT V2 now ?

I know, normally not here, but a quick answer.
I have made a new LXC with the TTeck-Script, for the new V2. Then I have copied the data-directory from the old LXC (opt/zigbee2mqtt/data) to the new LXC and replaced there the folder with the copied one. After that, I shutdown the old one, gave the new LXC the same IP (only for me, to have the same IP) and restarted, after that all done and works ;-)

**Comment by kopaszsop** [comment]
hi,
is there a way to install the 1.42.0 version of z2m with this script? i've problems with some ikea hardware wich worked for me but with the 2.0.0 version they don't.

**Reply by MickLesk** [reply]
no, only newest 2.0.0+ supported.

**Reply by kopaszsop** [reply]
ok, thx

**Reply by tugsi** [reply]
Which IKEA Hardware has problems? I have some too and no problems.

**Reply by kopaszsop** [reply]
> Which IKEA Hardware has problems? I have some too and no problems.

thx.
it's the E1743 on/off button.
z2m doesn't gives any switchable/controllable einties over to homeassistant

**Reply by tugsi** [reply]
Ok I´ve seen. I use ioBroker, there is everything fine, over MQTT you get the zigbee2mqtt/E1743/action too, where you have false/true for on/off .
But I have a test enviroment from HA and there I can see, is only battery, lastseen, linkquality, update . 
It seems there is something changed in Z2M, maybe better post an issue there. For the meanwhile you can go over an MQTT-Broker if you want to use 2.0+

**Reply by kopaszsop** [reply]
> Ok I´ve seen. I use ioBroker, there is everything fine, over MQTT you get the zigbee2mqtt/E1743/action too, where you have false/true for on/off . But I have a test enviroment from HA and there I can see, is only battery, lastseen, linkquality, update . It seems there is something changed in Z2M, maybe better post an issue there. For the meanwhile you can go over an MQTT-Broker if you want to use 2.0+

i've found this and tried all the checkboxes under the hood but i only get on-off-up-down-release states as "event" in HA.
https://github.com/Koenkk/zigbee2mqtt/issues/25502#issuecomment-2571333478

so this problem affects the z2m and not this script.

**Reply by tugsi** [reply]
I´ve seen your post and have checked in my installation the legacy action box and now I have the Action Entities in HA
![image](https://github.com/user-attachments/assets/53ba7811-640c-401a-8b8d-94660623ba95)

**Comment by Alyosha777** [comment]
I have just completed this setup and frontend is working for me and I have the configuration.yaml set up. My need is to migrate from Zigbee2mqtt running in HA. I have mosquito running in HA still.
I'm having difficulty importing my backup. The backup from HA has configuration.yaml, database.db, coordinator_backup.json, and state.json. Editing these folders on the Proxmox container does not lead to my devices showing up in my new Zigbee2mqtt. Anyone have success importing devices? I'd prefer to avoid repairing all devices.

**Comment by cyberdie** [comment]
Hi. The z2m lxc is not showing the usb adapter in /dev/serial/by-id path but lsusb shows the adapter correctly.
How can i configure z2m with this information? ttyUSB0/1, ttyACM0/1 didn't work :-(

![image](https://github.com/user-attachments/assets/92c8a538-16ee-4fc3-a989-7904567f50ee)

**Reply by MickLesk** [reply]
execute this on main node

**Reply by cyberdie** [reply]
No /dev/serial path on main node....

![image](https://github.com/user-attachments/assets/ef8dbc3d-b520-45e9-89ab-75ca6bb2d836)

**Reply by MickLesk** [reply]
then you dont have an connected COM Device on your port. Or Your device is broken

**Reply by cyberdie** [reply]
The device is shown in lsusb (Texas Instruments CC2531....)

**Reply by MickLesk** [reply]
and the COM Port is not reachable. look this or ask in zigbee2mqtt discussions
https://forum.iobroker.net/topic/29761/gel%C3%B6st-cc2531-an-proxmox-an-lxc-durchreichen-scheitert/61?lang=en-GB

**Comment by MattSpeke** [comment]
Trying to update Z2M from v1 using ttekk script to v2 using the new script. I get the error below. How do I upgrade/migrate from the old version to the new verson?
![image](https://github.com/user-attachments/assets/9cb69fad-81ab-4d64-87db-17019d076f12)

**Reply by MickLesk** [reply]
Use search function, read our Changelog and you found it
